### PR TITLE
copyurl: add option to print resulting auto-filename

### PR DESCRIPTION
**Pull request by albertony**
#### What is the purpose of this change?

Added new option:

> -p, --print-filename   Print the resulting name from --auto-filename

Example usage (PowerShell):
```
$x = .\rclone.exe copyurl https://www.youtube.com/watch?v=SzuZnh91U8E C:\Temp -a -p
PS > Write-Host "Downloaded file: ${x}"
Downloaded file: watch
```

#### Was the change discussed in an issue or in the forum before?

https://github.com/rclone/rclone/issues/5094

#### Checklist

- [X] I have read the [contribution guidelines](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#submitting-a-pull-request).
- [X] I have added tests for all changes in this PR if appropriate.
- [X] I have added documentation for the changes if appropriate.
- [X] All commit messages are in [house style](https://github.com/rclone/rclone/blob/master/CONTRIBUTING.md#commit-messages).
- [X] I'm done, this Pull Request is ready for review :-)

### CODE REVIEW

File: cmd/copyurl/copyurl.go [hunk 1]
```diff
@@ -13,15 +14,17 @@ import (
...
+	printFilename = false
+	stdout        = false
+	noClobber     = false
 )
 
 func init() {
 	cmd.Root.AddCommand(commandDefinition)
 	cmdFlags := commandDefinition.Flags()
 	flags.BoolVarP(cmdFlags, &autoFilename, "auto-filename", "a", autoFilename, "Get the file name from the URL and use it for destination file path")
+	flags.BoolVarP(cmdFlags, &printFilename, "print-filename", "p", printFilename, "Print the resulting name from --auto-filename")
```
Comment (ncw) [inline cmd/copyurl/copyurl.go:27]: Just a note for the future - we have to be very careful with short flags in sub commands - this means we can't use `-p` as a global flag... In `rsync` it is

        -p, --perms                 preserve permissions

So probably not much use!

We also use `-p` in `rclone tree` also.

I think this use is OK though.

File: cmd/copyurl/copyurl.go [hunk 2]
```diff
@@ -61,10 +65,14 @@ output to be written to standard output.
...
 		}
 		cmd.Run(true, true, command, func() error {
+			var dst fs.Object
 			if stdout {
 				err = operations.CopyURLToWriter(context.Background(), args[0], os.Stdout)
 			} else {
-				_, err = operations.CopyURL(context.Background(), fsdst, dstFileName, args[0], autoFilename, noClobber)
+				dst, err = operations.CopyURL(context.Background(), fsdst, dstFileName, args[0], autoFilename, noClobber)
+				if printFilename && err == nil && dst != nil {
+					fmt.Printf("%s", dst.Remote())
```
Comment (ncw) [inline cmd/copyurl/copyurl.go:74]: I think you want to print a newline here too - that is the unix way. So something like

    fmt.Println(dst.Remote())

**Review by ncw — COMMENTED** [review]
Looks good! One small query inline.

File: cmd/copyurl/copyurl.go (hunk 2 above)
Comment (albertony) [reply cmd/copyurl/copyurl.go]: Sure. I had it, amended it away, and now amended it back in! :)

File: cmd/copyurl/copyurl.go (hunk 1 above)
Comment (albertony) [reply cmd/copyurl/copyurl.go:27]: Oh, I did not think of that, thanks for the reminder. I can change it if you think that is best, but I were unable to come up with better alternative so no change in latest update... 👎

**Pull request merged**